lib/filewatch/watch.rb

```diff
@@ -155,6 +188,21 @@ def subscribe(stat_interval = 1, discover_interval = 5, &block)
     end # def subscribe
 
     private
+    def expired?(stat, watched_value)
+      return false unless expiry_enabled?
+      watched_value.size == stat.size && file_expired?(stat)
+    end
+
+    def discover_expired?(stat)
+      return false unless expiry_enabled?
+      file_expired?(stat)
+    end
+
+    def file_expired?(stat)
+      Time.now.to_i > (stat.mtime.to_i + @ignore_after)
```

Comment:
I am always a bit wary of operations on Time objects especially when there are hidden operations that could involve timezones (I have been bitten sooo many times by timezone issues).

However, just now, I looked at the JRuby code (I do this quite often to see exactly what is happening under the hood) and in RubyTime the `opMinus` method converts LHS and RHS to ints then floats before subtraction and returns a float. Then, in RubyFloat the `op_gt` method converts the RHS to a Java double before comparing it to the RubyFloat's internal double value.

Phew, a lot of conversions, I think using integers is more performant.
But I agree that the line is a bit WTF and violates POLS.